### PR TITLE
python3Packages.transformers: 2.2.1 -> 3.0.1

### PR DESCRIPTION
**Pull request by danieldk**
###### Motivation for this change

Changelog:
https://github.com/huggingface/transformers/releases/tag/v3.0.0

Also add myself as a maintainer.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc: @pashashocky

### CODE REVIEW

**Comment by danieldk** [comment]
@ofborg build python37Packages.transformers python38Packages.transformers

File: pkgs/development/python-modules/transformers/default.nix [hunk 1]
```diff
@@ -1,41 +1,76 @@
...
+    regex
+    requests
+    sacremoses
+    sentencepiece
+    tokenizers
+    tqdm
+  ];
+
+  checkInputs = [
+    pytest
```
Comment (drewrisinger) [inline pkgs/development/python-modules/transformers/default.nix:41]:
```suggestion
    pytestCheckHook
```

File: pkgs/development/python-modules/transformers/default.nix [hunk 2]
```diff
@@ -1,41 +1,76 @@
...
+, filelock
 , regex
+, requests
 , numpy
+, sacremoses
+, sentencepiece
+, timeout-decorator
+, tokenizers
+, tqdm
 , pytest
```
Comment (drewrisinger) [inline pkgs/development/python-modules/transformers/default.nix:14]:
```suggestion
, pytestCheckHook
```
I suggest using pytestCheckHook. Makes these disables cleaner.

File: pkgs/development/python-modules/transformers/default.nix [hunk 3]
```diff
@@ -1,41 +1,76 @@
...
+       not test_config_model_type_from_model_identifier and \
+       not test_from_pretrained_use_fast_toggle and \
+       not test_hf_api and \
+       not test_pretokenized_tokenizers and \
+       not test_tokenizer_equivalence_en_de and \
+       not test_tokenizer_from_model_type and \
+       not test_tokenizer_from_model_type and \
+       not test_tokenizer_from_pretrained and \
+       not test_tokenizer_identifier_with_correct_config"
   '';
```
Comment (drewrisinger) [inline pkgs/development/python-modules/transformers/default.nix:67]: Replace with (using ``pytestCheckHook``):
```
preCheck = "export HOME=$TMPDIR";
disabledTests = [
  "test_all_tokenizers"
  "test_batch_encoding_is_fast"
  ...
];

**Review by drewrisinger — CHANGES_REQUESTED** [review]
Suggest using pytestCheckHook for disabling these tests.

File: pkgs/development/python-modules/transformers/default.nix (hunk 2 above)
Comment (danieldk) [reply pkgs/development/python-modules/transformers/default.nix]: Great idea! Looks much better now, thanks!

**Review by drewrisinger — CHANGES_REQUESTED** [review]
* [x] Diff LGTM
* [x] Commits: Squash add maintainer. Remove tokenizers when #91342 is approved
* [x] Builds cleanly
* [x] DOESN'T run. See log below:

python38.transformers: Willing to accept, maybe include pytorch/tensorflow in inputs?? No strong opinion. (also, the spacing in this error message makes me want to file a PR... grr :D )
```
>>> import transformers
Neither PyTorch nor TensorFlow >= 2.0 have been found.Models won't be available and only tokenizers, configurationand file/data utilities can be used.
```

``python37Packages.transformers``: getting the same issue when launched from command-line or via ``transformers-cli``. Not sure what's causing this, seems like it's trying to use py38.numpy for some reason??

```
[nix-shell:~/.cache/nixpkgs-review/pr-91915/results/python37Packages.transformers]$ ./bin/transformers-cli
Traceback (most recent call last):
  File "/nix/store/0ryfy7j7azg5sgnh75sxlxbkbnd9vvbj-python3.8-numpy-1.18.5/lib/python3.8/site-packages/numpy/core/__init__.py", line 24, in <module>
    from . import multiarray
  File "/nix/store/0ryfy7j7azg5sgnh75sxlxbkbnd9vvbj-python3.8-numpy-1.18.5/lib/python3.8/site-packages/numpy/core/multiarray.py", line 14, in <module>
    from . import overrides
  File "/nix/store/0ryfy7j7azg5sgnh75sxlxbkbnd9vvbj-python3.8-numpy-1.18.5/lib/python3.8/site-packages/numpy/core/overrides.py", line 7, in <module>
    from numpy.core._multiarray_umath import (
ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/nix/store/rsxk262fy8xv1dsmpqb6klbw4sf1q4hf-python3.7-transformers-3.0.0/bin/.transformers-cli-wrapped", line 6, in <module>
    from transformers.commands.transformers_cli import main
  File "/nix/store/8ndbww1p3x936vjcxrjawa112wkvgc15-python3.8-transformers-3.0.0/lib/python3.8/site-packages/transformers/__init__.py", line 48, in <module>
    from .data import (
  File "/nix/store/8ndbww1p3x936vjcxrjawa112wkvgc15-python3.8-transformers-3.0.0/lib/python3.8/site-packages/transformers/data/__init__.py", line 6, in <module>
    from .processors import (
  File "/nix/store/8ndbww1p3x936vjcxrjawa112wkvgc15-python3.8-transformers-3.0.0/lib/python3.8/site-packages/transformers/data/processors/__init__.py", line 5, in <module>
    from .glue import glue_convert_examples_to_features, glue_output_modes, glue_processors, glue_tasks_num_labels
  File "/nix/store/8ndbww1p3x936vjcxrjawa112wkvgc15-python3.8-transformers-3.0.0/lib/python3.8/site-packages/transformers/data/processors/glue.py", line 24, in <module>
    from ...tokenization_utils import PreTrainedTokenizer
  File "/nix/store/8ndbww1p3x936vjcxrjawa112wkvgc15-python3.8-transformers-3.0.0/lib/python3.8/site-packages/transformers/tokenization_utils.py", line 26, in <module>
    from .tokenization_utils_base import (
  File "/nix/store/8ndbww1p3x936vjcxrjawa112wkvgc15-python3.8-transformers-3.0.0/lib/python3.8/site-packages/transformers/tokenization_utils_base.py", line 30, in <module>
    import numpy as np
  File "/nix/store/0ryfy7j7azg5sgnh75sxlxbkbnd9vvbj-python3.8-numpy-1.18.5/lib/python3.8/site-packages/numpy/__init__.py", line 142, in <module>
    from . import core
  File "/nix/store/0ryfy7j7azg5sgnh75sxlxbkbnd9vvbj-python3.8-numpy-1.18.5/lib/python3.8/site-packages/numpy/core/__init__.py", line 50, in <module>
    raise ImportError(msg)
ImportError:

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.7 from "/nix/store/73iskwap4wblkwbq2jy9cndmrn49n6ss-python3-3.7.7/bin/python3.7"
  * The NumPy version is: "1.18.5"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: No module named 'numpy.core._multiarray_umath'
```

**Comment by danieldk** [comment]
>     * [ ]  Commits: Squash add maintainer. Remove tokenizers when #91342 is approved

I think it's nice to have that in a commit separate from the version bump.

> python38.transformers: Willing to accept, maybe include pytorch/tensorflow in inputs?? No strong opinion. (also, the 

@pashashocky probably excluded these dependencies intentionally. `transformers` works with either, so I think it's up to the user to pick one. We use transformers with PyTorch and it would be nasty if we had to pull in Tensorflow as an additional dependency every time. Also Tensorflow has had quite a history of being broken frequently in `nixpkgs`.

> spacing in this error message makes me want to file a PR... grr :D )

:laughing: 

> `python37Packages.transformers`: getting the same issue when launched from command-line or via `transformers-cli`. Not sure what's causing this, seems like it's trying to use py38.numpy for some reason??

Let me check.

**Comment by danieldk** [comment]
> > `python37Packages.transformers`: getting the same issue when launched from command-line or via `transformers-cli`. Not sure what's causing this, seems like it's trying to use py38.numpy for some reason??
> 
> Let me check.

Binary seems fine?

```
(nix-build --no-out-link -A python3Packages.transformers)/bin/transformers-cli
Neither PyTorch nor TensorFlow >= 2.0 have been found.Models won't be available and only tokenizers, configurationand file/data utilities can be used.
usage: transformers-cli <command> [<args>]

positional arguments:
  {convert,download,env,run,serve,login,whoami,logout,s3,upload}
[...]
```

**Review by drewrisinger — APPROVED** [review]
Confirm that everything looks fine & builds. Approved pending removal of ``python3Packages.tokenizers`` commit from this PR.

**Comment by danieldk** [comment]
Rebased now that `python3Packages.tokenizers` is merged.

**Review by drewrisinger — APPROVED** [review]
* [x] Diff LGTM
* [x] Commits LGTM, minus subjective style of adding maintainer in separate commit
* [x] Builds clean via nix-review
* [x] tested via pytest & trivially w/ transformer-cli (same issue with python37Packages.transformers-cli as before, maybe re-test with that command & python37Packges.transformers in your test, vs python3Packages.transformers).

**Comment by danieldk** [comment]
>     * [x]  tested via pytest & trivially w/ transformer-cli (same issue with python37Packages.transformers-cli as before, maybe re-test with that command & python37Packges.transformers in your test, vs python3Packages.transformers).

Ah, I see the problem you are running into. If you run `nix-review`, the Python 3.8 modules end up in `PYTHONPATH`. Try to run `nix-review` as follows: `nix-review pr 91915 -p python37Packages.transformers`

Then `transformers-cli` should run as expected.

**Comment by danieldk** [comment]
Updated to transformers 3.0.1.

**Comment by drewrisinger** [comment]
@danieldk yeah, I figured something like that was happening, but I didn't have time to debug it.

**Pull request merged**